### ISSUE TITLE
Link to Gazetteer in main menu and on "Your actions" page have an accessibility issue

### ISSUE THREAD

**Issue by AlOneill** [FreeUKGen/FreeCENMigration#1462]
The information that the Gazetteer will open in a new tab is buried in a `title` attribute which means that any user who cannot hover (because they cannot or do not use a pointing device) will never see that information. The behaviour could surprise or even confuse them.

This issue will probably be deemed of low priority, but it is our stated aim to make our sites and pages as accessible as possible to all users. The user limitations that are relevant here should not make it more difficult for a volunteer, whether transcriber or coordinator or other role, to use the site comfortably.

**Comment by AlOneill** [comment]
I can suggest two possible approaches here:

1. incorporate some text into the main menu link to indicate that it will open in a new tab, e.g. "GAZETTEER (new tab)" — as this link is for members only I suggest that they will (soon!) understand that 'new tab' means that the link will open in a new tab

2. remove `target="_blank"` from the link and rely on the user to open the link in a new tab (if desired) themselves

As either solution would require an adjustment in mindset for many members, we would need to explain the change for them.

Which approach would the Scrum prefer? Can anyone propose a different (better for everyone?) approach?

**Comment by AlOneill** [comment]
Please note that a similar situation applies to the Gazetteer link (it looks like a button but it is a link) on the "Your actions" page. A similar solution is needed, i.e. set a visible (to all) warning or remove `target="_blank"`.

**Comment by DeniseColbert** [comment]
Option 2 is what we're to moving towards so for consistency we'll go with that.

i.e. remove target="_blank" from the link and rely on the user to open the link in a new tab (if desired) themselves

**Comment by AlOneill** [comment]
@DeniseColbert 
I have reviewed the link to the gazetteer on the Actions page today and note that:

- the link now opens in the _same_ tab

- however, the **title attribute** remains (bad) and it still says that the link will open in a _new_ tab
- the **title element** for the Gazetteer opening screen (which labels the browser tab) is simply 'FreeCEN |' (wrong: it should be 'Gazetter | FreeCEN') 

The main menu item opens in a new tab but the warning is inaccessible to keyboard-only users.

**Comment by DeniseColbert** [comment]
@PatReynoldsFUG to test

**Comment by PatReynoldsFUG** [comment]
In test 3:
"Button" gazatteer still says (when hovering) "opens in a new tab"

Browser tab still says  'FreeCEN |' 

Same on both Test3 and live

**Comment by DeniseColbert** [comment]
@Vino-S to make these changes (Alison's comment from November)

**Comment by geoffj-FUG** [comment]
If this is going to be a choice in keeping with policy can we add a rollover to this tab that says 'right click to open in new tab' for the Gazetteer menu items please.
This should stop a load of complaints from transcribers.
Geoff

**Comment by Vino-S** [comment]
Ready for testing in test3

**Comment by DeniseColbert** [comment]
@PatReynoldsFUG and @geoffj-FUG to test

**Comment by AlOneill** [comment]
@geoffj-FUG
It would not make sense to replace "Opens in new tab" with your suggestion as both are inaccessible to keyboard-only users (who cannot hover). 

We should be prepared for the complaints and respond with the reason. Members will get the idea soon enough.

**Comment by AlOneill** [comment]
I see that the Gazetteer link in the main menu still has a title attribute and that it reads on hover: "Search for an existing place name; Right click to open in new tab".

I hope this is temporary as **title attributes are not accessible to a keyboard-only user.**

**Comment by DeniseColbert** [comment]
To ask the Accessibility team re Alison's comments

**Comment by DeniseColbert** [comment]
Solution suggested: 

Hover text should read "Open in new tab" 

Place a line of text on Gazetteer page WTTE: "You might want to have this page open in a different tab while you work. You can open it in a new tab from the navigation menu".

**Comment by AlOneill** [comment]
@DeniseColbert 
Can I remind you that "hover text" (aka text in a `title` attribute) is not accessible to any user who cannot hover for any reason: keyboard only, broken mouse, etc. The whole point of this issue is to eliminate "hover text". 

I see what you are getting at by replacing "opens" with "open" but I fear it will be misread as "opens" and that will cause confusion. I maintain that removing the `title` attribute (hover text) is a better solution.

**Comment by DeniseColbert** [comment]
Thanks Alison: agreed, let's remove reference to that on hover to avoid confusion.

**Comment by AlOneill** [comment]
@DeniseColbert 
A compromise might be a **_temporary_** `title` attribute that reads "Please open in a new tab" but it must be removed in a month or so.

**Comment by Vino-S** [comment]
@geoffj-FUG to add comment.

**Comment by geoffj-FUG** [comment]
Pat

 

I thought this was discussed at the last scrum and that some sort of consensus had been achieved.

 

I don’t support the Gazetteer opening in the same tab at all. The change has been made on test3 and I continually end up with the Gazetteer overwriting my work and I have to start again. . I have washed my hands of this.

 

It is not my issue to resolve. I have no jurisdiction over this.

 

Geoff

**Comment by Vino-S** [comment]
Scrum decided  to keep existing way: Open Gazetter in new tab
@Vino-S Remove the Gazetter implementation from test3

**Comment by AlOneill** [comment]
I am disappointed that the Scrum decided to **keep the Gazetteer link inaccessible** to anyone who either cannot hover (because they navigate by keyboard) or who does not understand what has happened where a link opens in a new tab.

I was reminded this very week that the default browser behaviour is to open the link but to keep the focus on the originating tab. That was a disorienting experience for me — I wondered what had happened, what had gone wrong. It then took some digging in the browser settings to be able to change the behaviour so that the focus moved to the newly opened tab.

I hope the Scrum will reconsider the decision and reverse this backward step in improving the Accessibility of FreeCEN.

**Comment by DeniseColbert** [comment]
Alison to Action with some explanation on the Gazetteer page

**Comment by AlOneill** [comment]
@Vino-S 
I see an empty link in the blue box in the sidebar as part of the heading "What has changed?":
`<a href="/system/resources/W1siZiIsIjIwMjQvMDEvMzEvOG5sY3B3ZHN2MV9GcmVlQ0VOMl9IYW5kYm9va19wYXJ0XzFBX0phbl8yMDI0LnBkZiJdXQ/FreeCEN2-Handbook-part-1A-Jan-2024.pdf?sha=3d7fa56104267490" title="Handbook 1A"></a>`
Unless you tell me otherwise, I will delete this empty link. (I assume it is left over from devising the "Make a suggestion to improve the handbook" link.) Thanks.

**Comment by AlOneill** [comment]
@Vino-S 
I see another issue: you have marked up the link to "Make a suggestion …" as an aside with the same ARIA markup as the blue box, giving the link the title of "What has changed?" — oops, make that "Links and tabs". I do not think this is appropriate and so I will mark up the link in a `div`, without the ARIA attribute.

**Comment by AlOneill** [comment]
I have updated a DRAFT Handbook page on t3 with an explanation that the Gazetteer page will continue to open in a new tab, for the convenience of volunteers. See the second paragraph of "Links and tabs > What has changed?" on [Draft FreeCEN Handbook](https://test3.freecen.org.uk/cms/draft-freecen-handbook)

**Comment by geoffj-FUG** [comment]
Alison

 

That looks good. 

 

Geoff

**Comment by AlOneill** [comment]
Draft page published on t3. It is sync-ready.

**Comment by DeniseColbert** [comment]
Should be deployed to production so closing
